samplesheets/views.py

```diff
@@ -885,6 +909,171 @@ def handle_alerts_deactivate(cls, irods_request, app_alerts=None):
                 )
             )
 
+    @classmethod
+    def accept_request(
+        cls,
+        irods_request,
+        project,
+        request,
+        timeline=None,
+        taskflow=None,
+        app_alerts=None,
+    ):
+        """
+        Accept an iRODS accept request.
+
+        :param irods_request: IrodsDataRequest object
+        :param project: Project object
+        :param request: Request object
+        :param timeline: TimelineAPI instance or None
+        :param taskflow: TaskflowAPI instance or None
+        :param app_alerts: AppalertsAPI instance or None
+        :raise: FlowSubmitException if taskflow submission fails
+        """
+        tl_event = None
```

Comment:
We should check and fail if the status of the request is already `ACCEPTED`. Accepting the same request multiple times doesn't make sense. This might happen if different users open the accept form in the UI or if the accept API view is called multiple times.

This should raise an exception and also update the timeline event with `FAILED` status and an appropriate description.

And yes, this seems to have been missing from the original implementation. It's a bigger deal now that we have the API views.